packages/anchor-listener/src/listener.ts

```diff
@@ -0,0 +1,97 @@
+import { ANCHOR_CONTRACT_ADDRESSES } from '@ceramicnetwork/anchor-contract'
+import type { Block, Provider } from '@ethersproject/providers'
+import type { CID } from 'multiformats/cid'
+import {
+  type Observable,
+  type RetryConfig,
+  firstValueFrom,
+  fromEvent,
+  map,
+  mergeMap,
+  scan,
+} from 'rxjs'
+
+import { createRootCIDsLoader, mapLoadBlock } from './loader.js'
+
+export type ProcessedBlockEvent = {
+  type: 'processed-block'
+  block: Block
+  roots: Array<CID>
+}
+
+export type ReorganizationEvent = {
+  type: 'reorganization'
+  latestProcessedBlock: string
```

Comment:
I'm not entirely sure what lastProcessedBlock and newBlock represent

Reply:
Here `newBlock` would provide the full `Block` object from ethers: https://docs.ethers.io/v5/api/providers/types/#providers-Block and `latestProcessedBlock` would be the block hash of the `processed-block` event previously emitted, but flagged as a reorganization because `newBlock.parentHash != latestProcessedBlock`, so the consumer should handle the reorganization as needed, if I understand the purpose?

We can provide different information if needed, for example should we extract the root CIDs from the new block in this case?

Reply:
In our case I only need the re-orged block number so this is great.